src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -198,7 +198,13 @@ public boolean isBiblatexMode() {
 
         // 4. BIB file directory
         getDatabasePath().ifPresent(dbPath -> {
-            String parentDir = dbPath.getParent().toAbsolutePath().toString();
+            assert dbPath != null : "dbPath is null";
+            Path parentPath = dbPath.getParent();
+            if( parentPath == null ) {
+                parentPath = Paths.get(System.getProperty("user.dir"));
```

Comment:
Hi, folks, thank you for the comments!
I think the if (parentPath != null){parentPath = Paths.get(System.getProperty("user.dir"));} fix is essential. The fix expresses a usual Unix/Windows CLI convention that invoking "jabref biblio.bib" is actually equivalent to "jabref ./biblio.bib" and "jabref $(pwd)/biblio.bib". Without this fix, running "jabref biblio.bib" and clicking the "Get fulltext" button in the "General" tab throws a "null pointer exception" in the searcher thread, and the main GUI thread just "hangs" forever without yielding any results (and without any error messages...). IMHO, this is a show stopper.

With the suggested fix, JabRef behaves as most users would expect -- it reads 'biblio.bib' in the current directory, and, if a database directory has to be used as a base for searching files, it uses the same current directory as a 'biblio.bib' base. Please note that when the directory is specified explicitely, as in 'jabref ~/papers/bibliography.bib', this workaround is not used.